scripts/destroy_unused_workspaces.py

```diff
@@ -0,0 +1,148 @@
+        print(f"Workspaces from Terraform: {workspaces}")
+        # Filter workspaces that contain "pr" and replace spaces
+        pr_workspaces = [
+            workspace.replace(" ", "")
+            for workspace in workspaces
+            if prefix in workspace.lower()
+        ]
+        return pr_workspaces
+    else:
+        print(f"Error listing workspaces: {stderr}")
+        return []
+
+
+def destroy_workspace(workspace_name, project_name, project_short_name):
+    command_select = f"terraform workspace select {workspace_name}"
+    tf_vars = (
+        f"-var=project_name={project_name} "
+        f"-var=project_short_name={project_short_name} "
+        f"-var=client_id= -var=client_secret= -var=keycloak_environment="
+    )
+    command_destroy = f"terraform destroy {tf_vars} -auto-approve"
+    command_delete = f"terraform workspace select default && terraform workspace delete {workspace_name}"
+
+    try:
+        # Command: terraform workspace select
+        print(command_select)
+        return_code_select, stdout_select, stderr_select = execute_terraform_command(
+            command_select
+        )
+        if return_code_select != 0:
+            print(f"Error executing select command for workspace {workspace_name}")
+            return return_code_select, stdout_select, stderr_select
+
+        # Command: terraform destroy
+        print(command_destroy)
+        return_code_destroy, stdout_destroy, stderr_destroy = execute_terraform_command(
+            command_destroy
+        )
+        if return_code_destroy != 0:
+            print(f"Error executing destroy command for workspace {workspace_name}")
+            return return_code_destroy, stdout_destroy, stderr_destroy
+
+        # Command: terraform workspace delete
+        print(command_delete)
+        return_code_delete, stdout_delete, stderr_delete = execute_terraform_command(
+            command_delete
+        )
+        if return_code_delete != 0:
+            print(f"Error executing delete command for workspace {workspace_name}")
+            return return_code_delete, stdout_delete, stderr_delete
+
+        return 0, "Destroyed successfully", None
+    except Exception as e:
+        return 1, None, str(e)
+
+
+def destroy_workspace_wrapper(workspace_name, project_name, project_short_name):
+    try:
+        # Retry the destroy command if it returns non-zero exit code
+        for _ in range(2):  # You can adjust the number of retries as needed
+            return_code, stdout, stderr = destroy_workspace(
+                workspace_name, project_name, project_short_name
+            )
+            if return_code == 0:
+                return workspace_name, "Destroyed successfully"
+            else:
+                print(f"Retrying destroy command for workspace: {workspace_name}")
+        return workspace_name, stderr
+    except Exception as e:
+        return workspace_name, str(e)
+
+
+def main():
```

Comment:
We should get these parameters from command line. `docopt` is the simplest way to do this. We are already using this in our repo. For example usage see https://github.com/NHSDigital/gp-connect-appointments-management-fhir/blob/master/scripts/calculate_version.py
You can get name of the projects, short-code, etc.